packit_service/worker/events/new_hotness.py

```diff
@@ -0,0 +1,102 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+from logging import getLogger
+from typing import Optional, Tuple
+from urllib.parse import urlparse
+
+from packit_service.models import ProjectReleaseModel, AbstractTriggerDbType
+from packit_service.worker.events import AbstractForgeIndependentEvent
+
+logger = getLogger(__name__)
+
+
+class NewHotnessUpdateEvent(AbstractForgeIndependentEvent):
+    def __init__(
+        self,
+        package_name: str,
+        version: str,
+        distgit_project_url: str,
+    ):
+        super().__init__()
+        self.package_name = package_name
+        self.version = version
+        self.project_url = distgit_project_url
+
+        self._upstream_repo_namespace_and_name: Optional[Tuple] = None
+
+    @property
+    def upstream_project_url(self) -> str:
+        return self.package_config.upstream_project_url
+
+    def get_repo_namespace_and_name(self) -> Optional[Tuple]:
+        path = urlparse(self.upstream_project_url).path
+        if not path:
+            return None, None
+        # remove leading /
+        path = path[1:]
+        path_split = path.rsplit("/", maxsplit=1)
+        if len(path_split) != 2:
+            return None, None
+        return path_split[0], path_split[1]
```

Comment:
We have a class for this..;)

https://github.com/packit/ogr/blob/67470efc7e29942ddcfa99ac653d225f16455a6e/ogr/parsing.py#L8

Reply:
great! I will replace it with this